sdk/ml/azure-ai-ml/azure/ai/ml/_utils/utils.py

```diff
@@ -415,7 +415,7 @@ def dump_yaml_to_file(
         try:
             cm = open(dest, "w", encoding=DefaultOpenEncoding.WRITE)
         except OSError as e:  # FileNotFoundError introduced in Python 3
-            msg = "No such file or directory: {}"
+            msg = "No such file or not a file path: {}"
```

Comment:
Should we create the file if not exists during dump? btw, there are other places with the same error message during load.

Reply:
It will create a file if not exist, but it will not create folder. For the load part, I think the error message is less confusing.